modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -0,0 +1,333 @@
+            tokenizer = data_config["tokenizer"]
+        else:
+            tokenizer = None
+
+        if "transformations" in data_config:
+            transform_list = data_config["transformations"]
+        else:
+            transform_list = []
+
+        self.dataloader_info = DataLoaderInfo(
+            self.pipeline_id,
+            dataset_id=data_config["dataset_id"],
+            num_dataloaders=training_config["dataloader_workers"],
+            batch_size=training_config["batch_size"],
+            bytes_parser=data_config["bytes_parser_function"],
+            transform_list=transform_list,
+            storage_address=f"{self.modyn_config['storage']['hostname']}:{self.modyn_config['storage']['port']}",
+            selector_address=f"{self.modyn_config['selector']['hostname']}:{self.modyn_config['selector']['port']}",
+            num_prefetched_partitions=num_prefetched_partitions,
+            parallel_prefetch_requests=parallel_prefetch_requests,
+            tokenizer=tokenizer,
+        )
+
+    def _init_model_wrapper(self):
+        assert self.pipeline_id is not None
+        assert self.pipeline_config is not None
+        assert self.modyn_config is not None
+        assert self.base_dir is not None
+
+        self.model_wrapper = ModynModelWrapper(
+            self.modyn_config,
+            self.pipeline_id,
+            self.pipeline_config["training"]["device"],
+            self.base_dir,
+            f"{self.modyn_config['model_storage']['hostname']}:{self.modyn_config['model_storage']['port']}",
+        )
+
+    def _create_dirs(self):
+        assert self.pipeline_id is not None
+        assert self.base_dir is not None
+
+        # self.exp_output_dir = self.base_dir / f"{self.dataloader_info.dataset_id}_{self.pipeline_id}"
+        # self.drift_dir = self.exp_output_dir / "drift"
+        # os.makedirs(self.drift_dir, exist_ok=True)
+        # self.debug_dir = self.exp_output_dir / "debug"
+        # os.makedirs(self.debug_dir, exist_ok=True)
+
+    def init_data_drift_trigger(
+        self, pipeline_id: int, pipeline_config: dict, modyn_config: dict, base_dir: pathlib.Path
+    ):
+        self.pipeline_id = pipeline_id
+        self.pipeline_config = pipeline_config
+        self.modyn_config = modyn_config
+        self.base_dir = base_dir
+
+        self._init_dataloader_info()
+        self._init_model_wrapper()
+        self._create_dirs()
+
+    def inform_previous_trigger_and_model(
+        self, previous_trigger_id: Optional[int], previous_model_id: Optional[int]
+    ) -> None:
+        self.previous_trigger_id = previous_trigger_id
+        self.previous_model_id = previous_model_id
+        if previous_model_id is not None:
+            self.model_updated = True
+
+    def inform_previous_trigger_data_points(self, previous_trigger_id: int, data_points: int) -> None:
+        assert self.previous_trigger_id == previous_trigger_id
+        self.previous_data_points = data_points
```

Comment:
I think we should have two functions: 1) inform_model(model_id) which informs the trigger about the model id and 2) inform_trigger(trigger_id, num_datapoints) where we inform the trigger about a trigger (lol :D) and the number of data points. As mentioned somewhere else, this should also be in the common interface of trigger (can probably also be implemented there), and the other triggers can choose to ignore this, to avoid having many if statements checking the trigger type.

Reply:
done.